### PR TITLE
distilbert-sequence-classification

### PR DESCRIPTION
**Pull request by arunimaaaa**
Adding new notebook on NLP sequence classification

### CODE REVIEW

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+   ],
+   "source": [
+    "from transformers import AutoModelForSequenceClassification, AutoTokenizer\n",
+    "from openvino.inference_engine import IECore\n",
+    "import warnings\n",
+    "from pathlib import Path\n",
+    "import numpy as np\n",
+    "import onnx\n",
+    "import os\n",
+    "from urllib.parse import urlparse"
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:35]: I do not see this dependency used in notebook, probably can be removed?

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "from transformers import AutoModelForSequenceClassification, AutoTokenizer\n",
+    "from openvino.inference_engine import IECore\n",
+    "import warnings\n",
+    "from pathlib import Path\n",
+    "import numpy as np\n",
+    "import onnx\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:33]: there is no module usage

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+      "[ SUCCESS ] Total execution time: 4.55 seconds. \n",
+      "[ INFO ] The model was converted to IR v11, the latest model format that corresponds to the source DL framework input/output format. While IR v11 is backwards compatible with OpenVINO Inference Engine API v1.0, please use API v2.0 (as of 2022.1) to take advantage of the latest improvements in IR v11.\n",
+      "Find more information about API v2.0 and IR v11 at https://docs.openvino.ai\n"
+     ]
+    }
+   ],
+   "source": [
+    "onnx_model_path = 'model.onnx'\n",
+    "MODEL_DIR = f\"model/\"\n",
+    "os.makedirs(MODEL_DIR, exist_ok=True)\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:199]:
```suggestion
    "Path(MODEL_DIR).mkdir(exist_ok=True)\n",
```

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+      "NEGATIVE\n"
+     ]
+    }
+   ],
+   "source": [
+    "with open(\"data/sample.txt\",\"r\") as f:\n",
+    "    input_text = f.readlines()\n",
+    "print(\"User Input:\\n\",input_text[0])\n",
+    "result = infer(input_text)\n",
+    "print(result)"
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:368]: probably, if user will have several lines in file, it will break this sample, maybe it make sense to run in cycle line-by-line and print result for each line?

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "cc0c91a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def infer(input_text):\n",
+    "    input_text = tokenizer(input_text, padding='max_length', max_length = 512, truncation=True, return_tensors=\"pt\")\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:296]: do we really need such huge length size?
I think 128 should be pretty enough if you do not demonstrate model work on large texts.
Also, tokenizer is able to return numpy result if you change return_tensors value from `pt` to `np`, it is more native for openvino. If you wish to highlight that torch tensors are acceptable as input, probably some additional comments needed as highlighting for this behaviour

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+    "    inputs = {'input_ids': input_text['input_ids'], 'attention_mask': input_text['attention_mask']}\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:297]: probably you do not need repack, because tokenizer returns data in this format

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BATCH_SIZE = 1\n",
+    "MAX_SEQ_LENGTH = 512\n",
+    "\n",
+    "warnings.filterwarnings(\"ignore\")\n",
+    "\n",
+    "ie = IECore()\n",
+    "net_onnx = ie.read_network(model=ir_model_xml, weights=ir_model_bin)\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:238]: it is preferable to simplify reading to pass xml only
```suggestion
    "net_onnx = ie.read_network(ir_model_xml)\n",
```

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+   "id": "026c9eb5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BATCH_SIZE = 1\n",
+    "MAX_SEQ_LENGTH = 512\n",
+    "\n",
+    "warnings.filterwarnings(\"ignore\")\n",
+    "\n",
+    "ie = IECore()\n",
```
Comment (eaidova) [inline notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:237]: could you please use OpenVINO API 2.0?

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+    }
+   ],
+   "source": [
+    "onnx_model_path = 'model.onnx'\n",
+    "MODEL_DIR = f\"model/\"\n",
+    "os.makedirs(MODEL_DIR, exist_ok=True)\n",
+    "\n",
+    "onnx_model_path = Path(MODEL_DIR)/onnx_model_path\n",
+    "\n",
+    "!mo --input_model $onnx_model_path --output_dir $MODEL_DIR --model_name $checkpoint --input input_ids,attention_mask --input_shape [1,512],[1,512] --data_type FP32"
```
[the remark this answers is not in the document]
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: image

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 6 above)
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:297]: if i pass the direct input without repack
image

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 7 above)
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:238]: Modified

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 8 above)
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:237]: Yes, modified

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 6 above)
Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:297]:
```suggestion
    "    inputs = dict(input_text)\n",
```

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 5 above)
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:296]: input_text = tokenizer(input_text, padding='max_length', max_length = 128, truncation=True, return_tensors="np")
image

Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:296]: yeah, probably these changes also requires update mo command line

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 9 above)
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: "!mo --input_model $onnx_model_path --output_dir $MODEL_DIR --model_name $checkpoint --input 'input_ids[1 512],attention_mask[1 512]'"

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 8 above)
Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:237]: but looks like not uploaded, I see only 1 commit here for now...

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 9 above)
[the remark this answers is not in the document]
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: !mo --input_model $onnx_model_path --output_dir $MODEL_DIR --model_name $checkpoint --input 'input_ids[1 128],attention_mask[1 128]'
image

This works.
!mo --input_model $onnx_model_path --output_dir $MODEL_DIR --model_name $checkpoint --input input_ids,attention_mask --input_shape [1,128],[1,128] --data_type FP32

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 8 above)
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:237]: yes, I'm testing. I'll upload after making all changes you're asking.

File: notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb (hunk 9 above)
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: which openvino version do you use?

[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: anyway please remove data type, it is not requires

[the remark this answers is not in the document]
Comment (arunimaaaa) [reply notebooks/224-distilbert-Sequence-Classification/224-distilbert-Sequence-Classification.ipynb:203]: I'm using 2022.2.0.dev20220829. Data type I removed
